src/main.go

```diff
@@ -127,25 +128,25 @@ func collectParamsFile(systemEntity *integration.Entity, argsParamLoc string) er
 // Debian/Ubuntu and RHEL/CentOS systems.
 func determineParamsFileLoc(argsParamLoc string) (*string, error) {
 	if argsParamLoc != "" {
-		if _, err := os.Stat(argsParamLoc); os.IsNotExist(err) {
+		if _, err := os.Stat(argsParamLoc); err != nil {
 			return nil, err
 		}
 		return &argsParamLoc, nil
 	}
 
 	// Try Debian/Ubuntu path
 	paramsLoc := debianUbuntuParamsLoc
-	if _, err := os.Stat(paramsLoc); !os.IsNotExist(err) {
+	if _, err := os.Stat(paramsLoc); err == nil {
 		return &paramsLoc, nil
 	}
 
 	// Try RHEL/CentOS
 	paramsLoc = rhelCentosParamsLoc
-	if _, err := os.Stat(paramsLoc); !os.IsNotExist(err) {
+	if _, err := os.Stat(paramsLoc); err == nil {
 		return &paramsLoc, nil
 	}
 
-	return nil, errors.New("varnish.params file could not be found")
+	return nil, fmt.Errorf("%w: %s or %s", errParamsFileNotFound, debianUbuntuParamsLoc, rhelCentosParamsLoc)
```

Comment:
Returning a errParamsFileNotFound  is not 100% correct. 
If we reach this point we are not sure why we had an error for `debianUbuntuParamsLoc` and `rhelCentosParamsLoc`. It could be that the file does not exist or that we did not have enough permissions to verify that

Reply:
maybe it is better just to provide a generic error saying that we were not able to read such file

Reply:
Yes, This makes sense to me. I will correct it

Reply:
We are not actually reading the file over here right? Just trying to check if file is present at any of known locations or not? I feel `could not access` may be the right word, What do you think?

 errors.New("could not access the varnish.params file")

Reply:
Yes we are simply validating its existence, however if we fail doing so we cannot assume it is because it does not exist or because we do not have enough power to do so.

I like your new wording!

Reply:
No problem at all to improve further 😄! I've added the debug log to print the reasons for errors